.circleci/config.yml

```diff
@@ -1,60 +1,172 @@
+#
+#  Globality autogenerated CircleCI configuration
+#
+#  This file is auto generated with globality-build.
+#  You should not make any changes to this file manually
+#
+#  Any changes made to this file will be overwritten in the
+#  next version of the build.
+#
+#  See: http://github.com/globality-corp/globality-build
+#
+#
+
+defaults: &defaults
+  working_directory: ~/repo
+  docker:
+    - image: ${AWS_ECR_DOMAIN}/globality-build:stable
+      aws_auth:
+        aws_access_key_id: ${AWS_ACCESS_KEY_ID}
+        aws_secret_access_key: ${AWS_SECRET_ACCESS_KEY}
+      environment:
+        EXTRA_INDEX_URL: "InjectedDuringRuntime"
+        AWS_ECR_DOMAIN: "InjectedDuringRuntime"
+        JFROG_AUTH: "InjectedDuringRuntime"
+        PYPI_USERNAME: "InjectedDuringRuntime"
+        PYPI_PASSWORD: "InjectedDuringRuntime"
+
+deploy_defaults: &deploy_defaults
+  working_directory: ~/repo
+  docker:
+    - image: ${AWS_ECR_DOMAIN}/globality-build:stable
+      aws_auth:
+        aws_access_key_id: ${AWS_ACCESS_KEY_ID}
+        aws_secret_access_key: ${AWS_SECRET_ACCESS_KEY}
+      environment:
+        EXTRA_INDEX_URL: "InjectedDuringRuntime"
+        AWS_ECR_DOMAIN: "InjectedDuringRuntime"
+        JFROG_AUTH: "InjectedDuringRuntime"
+        PYPI_USERNAME: "InjectedDuringRuntime"
+        PYPI_PASSWORD: "InjectedDuringRuntime"
+
+whitelist: &whitelist
+  paths:
+    .
+
 version: 2
 
 jobs:
+  checkout:
+    <<: *defaults
 
-  build-python-library:
-    docker:
-      - image: circleci/python:3.6.2
     steps:
       - checkout
+
+      - persist_to_workspace:
+          root: ~/repo
+          <<: *whitelist
+
+  build_base_docker:
+    <<: *defaults
+
+    steps:
+      - attach_workspace:
+          at: ~/repo
+
+      - setup_remote_docker
+
       - run:
-          name: Install Pre-Dependencies
-          command: sudo pip install tox
-      - restore_cache:
-          key: v1-python-dependencies-{{ checksum "setup.py" }}
+          name: Build Base Docker
+          # install dependencies for loading ecs task definitions
+          command: |
+            eval $(aws ecr get-login --no-include-email)
+            python -m middlefield globality build build-gen --config .globality/build.json
```

Comment:
Why are we generating the build here?

Reply:
Generating the build here means that we can't guarantee that build 171 and build 172 use the same build configuration.

Reply:
it means if we upgrade globality-build, we don't need a loop of updates to all projects.